### PR TITLE
[lib] make canonical_path_name always absolute (fix #13031)

### LINKED ISSUE #13031: coq_makefile -destination-of dir/F fails if dir exists but does not contain F
#### Description of the problem

Ran into a bit of a bizarre bug where `coq_makefile -destination-of` prints the correct output *unless* the top-level directory exists, in which case it prints nothing. I tested it out by creating an empty directory, trying `coq_makefile -destination_of` with dummy arguments, and then running `mkdir src` and running it again. The second run produces no output. (I also tested it out with directory names other than `src` in case that one was special, and the issue is the same.)

```
$ ls -a
.  ..
$ coq_makefile -destination-of src/Y/Z/Test.v -Q src X
X//Y/Z
$ mkdir src
$ coq_makefile -destination-of src/Y/Z/Test.v -Q src X
```

This is causing `make install` to fail for me on a few different projects that rely on `-destination-of`.

#### Coq Version

```
$ coqtop -v
The Coq Proof Assistant, version 8.12.0 (September 2020)
compiled on Sep 15 2020 14:27:23 with OCaml 4.08.1
```

### PR DESCRIPTION
**Pull request by gares**
Fix #13031

### CODE REVIEW

**Comment by gares** [comment]
I'm shocked nothing broke... well, anyone taking this?

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
The error handling clause is probably never triggered in a typical workflow. #13031 is a bit peculiar in this respect.

File: clib/cUnix.ml [hunk 1]
```diff
@@ -69,7 +69,7 @@ let canonical_path_name p =
     p'
   with Sys_error _ ->
     (* We give up to find a canonical name and just simplify it... *)
-    strip_path p
+    current ^ dirsep ^ strip_path p
```
Comment (SkySkimmer) [inline clib/cUnix.ml:72]: Are we never called with an absolute path that doesn't start with PWD? Like `/doesnotexist` or some such thing.
Also stripping a leading PWD just to put it back is kinda strange.

Comment (gares) [reply clib/cUnix.ml:72]: Fair enough, but it is hard to tell what a function is supposed to do when it is not documented.

If you want to add a test for a leading / you are welcome to do so. Note that strip_path also gets rid of stuff like `/./`, so you should not just return `p` IMO.

**Comment by ppedrot** [comment]
Merging anytime soon, @SkySkimmer are you OK?

**Comment by SkySkimmer** [comment]
not opposing

**Pull request merged**